src/SourceBuild/content/prep.sh

```diff
@@ -171,6 +176,32 @@ function BootstrapArtifacts {
   rm -rf "$workingDir"
 }
 
+function RemoveBinaries {
+  echo "  Removing binaries..."
+
+  # Define the paths to ignore when removing binaries
+  pathsToIgnore=(
+    "*.dotnet*"
+    "*artifacts*"
+    "**/eng/tools/**/bin/**"
+    "**/eng/tools/**/obj/**"
+    "$SCRIPT_ROOT/prereqs/packages/**"
+    "$SCRIPT_ROOT/test/Microsoft.DotNet.SourceBuild.SmokeTests/bin/**"
+    "$SCRIPT_ROOT/test/Microsoft.DotNet.SourceBuild.SmokeTests/obj/**"
+    "$SCRIPT_ROOT/test/Microsoft.DotNet.SourceBuild.SmokeTests/TestResults/**"
+  )
+
+  # Define the find command
+  findCmd="find \"$SCRIPT_ROOT\" -type f"
+  for path in "${pathsToIgnore[@]}"; do
+    findCmd+=" -not \( -path \"$path\" \)"
+  done
+  findCmd+=" -regex '.*\.\(dll\|Dll\|exe\|mdb\|nupkg\|pdb\|tgz\|zip\)$' -delete -print"
```

Comment:
Why does this only search for these file types? Shouldn't _all_ binary files be removed?

Reply:
Fixed to remove all binaries https://github.com/dotnet/installer/pull/18541/commits/bc537bea7b68cadbcab8284e148a9e9c9241b9e3